### ISSUE TITLE
nautilus plugin cannot share folders

### ISSUE THREAD

**Issue by individual-it** [owncloud/client#3556]
Version: nightly build of 2.0, build from GIT-Revision db8b91
OS: Linux deepthought2 3.16-3-amd64 #1 SMP Debian 3.16.5-1tanglu1 (2014-10-13) x86_64 GNU/Linux

When right clicking on a file within the OC folder in nautilus, I get the "share with Owncloud" menu entry. But on a folder it does not work.

Output when starting nautilus from the console
```
Socket: /run/user/1000/ownCloud/socket <=> /ownCloud/socket
Socket File: /run/user/1000/ownCloud/socket
Setting connected to True
Socket watch id: 18
Initializing nautilus-open-terminal extension
Initializing nautilus-dropbox 2015.02.12
Initializing nautilus-image-converter extension
Server response: REGISTER_PATH:/home/user/OwnCloud/account1/
Server response: REGISTER_PATH:/home/user/OwnCloud/account2/
Server response: STATUS:OK:/home/user/OwnCloud/account1/
Server response: STATUS:OK:/home/user/OwnCloud/account2/
Server response: STATUS:OK:/home/user/OwnCloud/account2/Secure/
Server response: STATUS:OK:/home/user/OwnCloud/account2/ICT/
Server response: STATUS:OK:/home/user/OwnCloud/account2/Photos/
Server response: STATUS:OK:/home/user/OwnCloud/account2/Documents/
Server response: STATUS:OK+SWM:/home/user/OwnCloud/account2/Docs/
Server response: STATUS:OK+SWM:/home/user/OwnCloud/account2/testshare/
Server response: STATUS:OK:/home/user/OwnCloud/account2/areas_of_ICT_work.xls
Server response: STATUS:IGNORE:/home/user/OwnCloud/account2/.owncloudsync.log
Server response: STATUS:OK:/home/user/OwnCloud/account2/music/
Server response: STATUS:OK:/home/user/OwnCloud/account2/ownCloudUserManual.pdf
Server response: STATUS:IGNORE:/home/user/OwnCloud/account2/.csync_journal.db-shm
Server response: STATUS:IGNORE:/home/user/OwnCloud/account2/.csync_journal.db-wal
Server response: STATUS:IGNORE:/home/user/OwnCloud/account2/.csync_journal.db
Traceback (most recent call last):
  File "/usr/share/nautilus-python/extensions/syncstate.py", line 163, in get_file_items
    if filename.startswith(reg_path) and socketConnect.nautilusVFSFile_table[filename]['state'] == 'OK':
KeyError: '/home/user/OwnCloud/account2/ICT'
Traceback (most recent call last):
  File "/usr/share/nautilus-python/extensions/syncstate.py", line 163, in get_file_items
    if filename.startswith(reg_path) and socketConnect.nautilusVFSFile_table[filename]['state'] == 'OK':
KeyError: '/home/user/OwnCloud/account2/Docs'
```

**Comment by rullzer** [comment]
@dragotin didn't you write the nautilus stuff?

**Comment by individual-it** [comment]
just played around with the python code a little bit
the problem in in line 163 of syncstate.py as the error says.
the table "socketConnect.nautilusVFSFile_table" does not have an entry [filename] for the folders. because in this table the folders are listed with an "/" at the end
so this line would work for folders:

    if filename.startswith(reg_path) and socketConnect.nautilusVFSFile_table[filename+"/"]['state'] == 'OK':

but of course not anymore for files
Would need to have a check if "filename" is a file or folder and add a "/" to the "filename" if its a folder

And even if I make it work with folders I get an Error 403 when trying to share them, not sure if that is connected

**Comment by individual-it** [comment]
fixed by #3601
